### PR TITLE
Added metadata_autotag and backup_database

### PR DESCRIPTION
**Pull request by David-Maisonave**
metadata_autotag has same interface as metadata_clean.
backup_database takes no arguments.

### CODE REVIEW

File: stashapp.py [hunk 1]
```diff
@@ -336,6 +336,28 @@ def metadata_clean(self, paths:list=[], dry_run=False):
 		result = self.call_GQL(query, {"input": clean_metadata_input})
 		return result
 
+	def metadata_autotag(self, paths:list=[], dry_run=False):
```
Comment (stg-annon) [inline stashapp.py]: we would either want to accept all possible values for `AutoTagMetadataInput` or just accept a dict so the user can pass the whole input themselves

**Review by stg-annon — CHANGES_REQUESTED** [review]
thanks for the contribution but this should probably be a separate PR from 
> "Added metadata_autotag and backup_database"

File: stashapp.py (hunk 1 above)
Comment (David-Maisonave) [reply stashapp.py]: > we would either want to accept all possible values for AutoTagMetadataInput or just accept a dict so the user can pass the whole input themselves

How would that be implemented?

> thanks for the contribution but this should probably be a separate PR from

I agree.  I didn't meant for the template  changes to get bunch together with these changes.  I'll pull the template changes out, and wait until this goes through.

Comment (stg-annon) [reply stashapp.py]: for example
```py
def metadata_autotag(self, paths:list=[], performers:list=[], stuios:list=[], tags:list=[]):
	metadata_autotag_input = {
		"paths":paths,
		"performers": performers,
		"studios":studios
		"tags":tags
	}
```
or 

```py
def metadata_autotag(self, input:dict={}):
	metadata_autotag_input = input
```
you can see the documentation for the api in the `/playground` in your stash
image



also just noticing the `dry_run` flag is not needed here

**Comment by David-Maisonave** [comment]
> also just noticing the dry_run flag is not needed here

Yes, this gave me problems when I was trying to implemented, because I tried to duplicate the code I saw in metadata_clean. It kept failing until I realized I had to remove **dry_run** from metadata_autotag_input.
I guess I forgot to remove it from the input arguments.

I'll update changes shortly.

**Comment by David-Maisonave** [comment]
I'm removing the check for paths being empty, since it works even if it's empty.
```
            if not paths:
                return
```

Why is this check in **metadata_clean**? When I remove it from that function, it also works with no input.
Do you have any issues with me removing it from both **metadata_autotag** and **metadata_clean** on this update?

**Comment by David-Maisonave** [comment]
> you can see the documentation for the api in the /playground in your stash

I guess I'm must be missing something obvious here, because I don't see how you got that screen display.
I did try populating the GraphiQL with the following, but got an error:
```
{
		mutation MetadataAutoTag($input:AutoTagMetadataInput!) {
		    metadataAutoTag(input: $input)
		}
}
```

How did you get that screen?

**Comment by David-Maisonave** [comment]
I figured out how to use the playground, and I now know what you mean my self documenting. But that doc button is tiny and it's in the upper right corner where it's easy to miss.

I recommend putting some screen shots in the ReadMe for [CommunityScripts ](https://github.com/stashapp/CommunityScripts) showing how to access the document.
Example Usage Instructions:
- Enter http://localhost:9999/playground on your browser, and click on the document button on the right side of the screen.
- GraphiQL_Doc1
- Click on the interested sub section ([Query](http://localhost:9999/playground#), [Mutation](http://localhost:9999/playground#), or [Subscription](http://localhost:9999/playground#))
- GraphiQL_Doc2
- That will display a list of documented API's.
- GraphiQL_Doc3
- Clicking on the input variables will display more details.

**Comment by David-Maisonave** [comment]
Added metadata_clean_generated.

I believe I made all the changes per review.
Please let me know if I missed anything.

**Comment by David-Maisonave** [comment]
Please let me know if there's no interest in this pull request so I can close it.

Thank you

**Review by stg-annon — APPROVED** [review]

**Pull request merged**